Px.Utils/Validation/PxFileValidator.cs

```diff
@@ -39,99 +37,159 @@ public void SetCustomValidatorFunctions(CustomSyntaxValidationFunctions? customS
             DataValidator dataValidator = new(
-                stream,
                 contentValidationResult.DataRowLength,
                 contentValidationResult.DataRowAmount,
-                filename,
                 syntaxValidationResult.DataStartRow, 
-                encoding,
                 syntaxConf);
-            ValidationResult dataValidationResult = dataValidator.Validate();
+            ValidationResult dataValidationResult = dataValidator.Validate(stream, filename, encoding, leaveStreamOpen, fileSystem);
             feedbacks.AddRange(dataValidationResult.FeedbackItems);
 
+            if (_customStreamValidators is not null)
+            {
+                foreach (IPxFileStreamValidator customValidator in _customStreamValidators)
+                {
+                    ValidationResult customValidationResult = customValidator.Validate(stream, filename, encoding, leaveStreamOpen, fileSystem);
+                    feedbacks.AddRange(customValidationResult.FeedbackItems);
+                }
+            }
             if (_customValidators is not null)
             {
-                foreach (IPxFileValidator customValidator in _customValidators)
+                foreach (IValidator customValidator in _customValidators)
                 {
                     ValidationResult customValidationResult = customValidator.Validate();
                     feedbacks.AddRange(customValidationResult.FeedbackItems);
                 }
             }
 
-            return new ValidationResult([..feedbacks]);
+            return new ValidationResult(feedbacks);
         }
 
         /// <summary>
         /// Validates the Px file asynchronously. Starts with metadata syntax validation, then metadata content validation, and finally data validation.
         /// If any custom validation functions are set, they are executed after the default validation steps.
+        /// <param name="stream">Px file stream to be validated</param>
+        /// <param name="filename">Name of the file subject to validation</param>
+        /// <param name="encoding">Encoding format of the px file</param>
+        /// <param name="leaveStreamOpen">Boolean value that determines whether the stream should be left open after validation.
+        /// <param name="fileSystem">File system to use for file operations. If not provided, default file system is used.</param>
+        /// <param name="cancellationToken">Cancellation token for cancelling the validation process</param>
         /// </summary>
         /// <returns><see cref="ValidationResult"/> object that contains the feedback gathered during the validation process.</returns>
-        public async Task<ValidationResult> ValidateAsync(CancellationToken cancellationToken = default)
+        public async Task<ValidationResult> ValidateAsync(
+            Stream stream,
+            string filename,
+            Encoding? encoding = null,
+            bool leaveStreamOpen = false,
```

Comment:
Remove leaveStreamOpen and make sure no references to it are kept in the class. Disposing the stream should be the responsibility of the caller.